nemo/collections/nlp/modules/common/megatron/retrieval_token_level_encoder_decoder.py

```diff
@@ -86,6 +86,7 @@ def __init__(
         enc_cross_attention=[3],  # layer numbers for cross attention
         dec_cross_attention=[3, 5],  # layer numbers for chunked cross attention
         add_position_embedding=False,
+        eod_id=None,  # end of sequence token id
```

Comment:
Can we pass a tokenizer object directly and then access whatever extra tokens we want from it like pad, eos. In the future, this may make it easier than passing specific token_ids.

Reply:
if we passing-in the tokenizer, it makes unit test difficult as we need to instantiate a tokenizer. Passing an id is light weighted.

Reply:
Instantiating a tokenizer is just one function call though. @ericharper wrote GPT unittests recently with tokenizer instantiation, maybe we could just re-use that part?

Reply:
I used a mock tokenizer in the test. It works this way